### PR TITLE
[tizen_package_manager] API cleanups and code refactoring

### PR DESCRIPTION
**Pull request by swift-kim**
- Rename `package_manager.dart` to `tizen_package_manager.dart`.
- Create an enum type `StorageType` for `PackageInfo.installedStorageType`.
- Change the return type of `PackageManager.install`/`uninstall`/`update` to `void`. They now throw a `PlatformException` in case of an error.
- Refactor (rewrite) the C++ code.
- Revise the example app and add a "Package events" screen. (c.f. tizen_app_manager's example app)
- General cleanups.

### CODE REVIEW

File: packages/tizen_package_manager/tizen/src/tizen_package_manager.h [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+class TizenPackageManager {
+ public:
+  ~TizenPackageManager();
+
+  // Returns a unique instance of TizenPackageManager.
+  static TizenPackageManager& GetInstance() {
+    static TizenPackageManager instance;
+    return instance;
+  }
+
```
[the remark this answers is not in the document]
Comment (swift-kim) [reply packages/tizen_package_manager/tizen/src/tizen_package_manager.h:45]: Done

File: packages/tizen_package_manager/tizen/src/tizen_package_manager.h [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+struct PackageInfo {
+  std::string package_id;
+  std::string label;
+  std::string type;
+  std::optional<std::string> icon_path;
+  std::string version;
+  std::string installed_storage_type;
+  bool is_system;
+  bool is_preloaded;
+  bool is_removable;
```
[the remark this answers is not in the document]
Comment (swift-kim) [reply packages/tizen_package_manager/tizen/src/tizen_package_manager.h]: Done

File: packages/tizen_package_manager/tizen/src/tizen_package_manager.cc [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+  package_manager_request_h request = nullptr;
+  int ret = package_manager_request_create(&request);
+  if (ret != PACKAGE_MANAGER_ERROR_NONE) {
+    LOG_ERROR("package_manager_request_create failed: %s",
+              get_error_message(ret));
+    last_error_ = ret;
+    return false;
+  }
+
+  ret = package_manager_request_set_type(request, "unknown");
```
[the remark this answers is not in the document]
Comment (swift-kim) [reply packages/tizen_package_manager/tizen/src/tizen_package_manager.cc:253]: The API is not documented very well, and I don't know what the call is for, so I just followed the original implementation. It worked well without specifying the type so I think we don't have to do so. Maybe we could even delete the call to `package_manager_request_set_type` if it's unnecessary.

[the remark this answers is not in the document]
Comment (bbrto21) [reply packages/tizen_package_manager/tizen/src/tizen_package_manager.cc:253]: Let's leave it as is.

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**